module/applications/actor/group-sheet.mjs

```diff
@@ -0,0 +1,299 @@
+      [movement.water, `${game.i18n.localize("DND5E.MovementWater")} ${movement.water}`],
+      [movement.air, `${game.i18n.localize("DND5E.MovementAir")} ${movement.air}`]
+    ];
+    speeds = speeds.filter(s => !!s[0]).sort((a, b) => b[0] - a[0]);
+    const primary = speeds.shift();
+    return {
+      primary: `${primary ? primary[1] : "0"}`,
+      secondary: speeds.map(s => s[1]).join(", ")
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare inventory items for rendering on the sheet.
+   * @returns {Object<string,object>}
+   */
+  #prepareInventory(items) {
+
+    // Categorize as weapons, equipment, containers, and loot
+    const sections = {
+      weapon: {label: "DND5E.ItemTypeWeaponPl", items: [], hasActions: false, dataset: {type: "weapon"}},
+      equipment: {label: "DND5E.ItemTypeEquipmentPl", items: [], hasActions: false, dataset: {type: "equipment"}},
+      backpack: {label: "DND5E.ItemTypeContainerPl", items: [], hasActions: false, dataset: {type: "backpack"}},
+      loot: {label: "DND5E.ItemTypeLootPl", items: [], hasActions: false, dataset: {type: "loot"}}
+    };
+
+    // Classify items
+    for ( const item of items ) {
+      const {quantity} = item.system;
+      item.isStack = Number.isNumeric(quantity) && (quantity > 1);
+      item.canToggle = false;
+      if ( ["weapon", "equipment", "backpack"].includes(item.type) ) sections[item.type].items.push(item);
+      else sections.loot.items.push(item);
+    }
+    return sections;
+  }
+
+  /* -------------------------------------------- */
+  /*  Rendering Workflow                          */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async _render(force, options={}) {
+    for ( const member of this.object.system.members) {
+      member.apps[this.id] = this;
+    }
+    return super._render(force, options);
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async close(options={}) {
+    for ( const member of this.object.system.members ) {
+      delete member.apps[this.id];
+    }
+    return super.close(options);
+  }
+
+  /* -------------------------------------------- */
+  /*  Event Listeners and Handlers                */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  activateListeners(html) {
+    super.activateListeners(html);
+    html.find(".group-member .name").click(this.#onClickMemberName.bind(this));
+
+    // Action buttons
+    html.find(".action-button").click(this.#onClickActionButton.bind(this));
+    html.find(".item-control").click(this.#onClickItemControl.bind(this));
+
+    // Item summaries
+    html.find(".item .rollable h4").click(event => this.#onClickItemName(event));
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Handle clicks to action buttons on the group sheet.
+   * @param {PointerEvent} event      The initiating click event
+   */
```

Comment:
This method isn't intended to have a meaningful return value, doing:
```js
return this.object.system.removeMember(removeMemberId);
```
Is just a more code-frugal approach than:
```js
this.object.system.removeMember(removeMemberId);
break;
```
If we want to be strict about our return types I will remove the return statements from the method so that it returns void.